### ISSUE TITLE
Partial toc problem

### ISSUE THREAD

**Issue by WenyinWei** [HarveySheppard/yLaTeX#11]
Hello Harvey,

I am grateful to use your beautiful typography cls file and have made some edits to it. But maybe I need some help of you to make the partial contents for chapter really partial. I have the problem like that 

![image](https://user-images.githubusercontent.com/33152225/50449471-91425800-0962-11e9-827b-0f0a8edab0d1.png)

The section in the chapter appears perfectly, but others do not belong to the chapter also appears, which is really annoying. I use such a snippet to print the partial table.

```
\chapter{Power balance in a fusion reactor}
\printMarginPartialToc

\section{Introduction}

Based on the results derived in Chapter 3 it is now possible to assemble all the sources and
```

And here is the part of my cls setting concerning table of contents.

Anticipate your reply and thanks for your help in advance.

Best wishes,
Wenyin 
```

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%		Table of content
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\usepackage{titletoc}
\newcommand{\yTableOfContent}[1][]{%
	\nomarginparpage
	\hypersetup{linkcolor=black}
	\setlength{\parskip}{2mm}
	\tableofcontents
	#1
	\newpage
	\marginparpage
}

\newcommand{\yTheContentsPage}{{\hypersetup{linkcolor=darkGrey},\quad\thecontentspage}}

\titlecontents*{chapter}
[2em] % Left
{\fontspec{Fira Sans SemiBold}} % Above code
{\contentslabel[\color{mainColor}\thecontentslabel]{2em}} % Numbered entry format
{\hspace*{-2em}} % numberless entry format
{\yTheContentsPage} % filler page format
[] % below code

\titlecontents{section}
[4em] % Left
{\fontspec{Fira Sans SemiBold}} % Above code
{\contentslabel{2em}} % Numbered entry format
{\hspace*{-2em}} % numberless entry format
{\yTheContentsPage} % filler page format
[] % below code

\titlecontents{subsection}
[6em] % Left
{} % Above code
{\contentslabel{3em}} % Numbered entry format
{\hspace*{-2em}} % numberless entry format
{\yTheContentsPage} % filler page format
[] % below code

\makeatletter
	% \pretocmd{\@schapter}{\startcontents[chapter]}{}{} % uncomment to activate mini toc in frontmatter
	\pretocmd{\@chapter}{\startcontents}{}{}
\makeatother

\apptocmd{\contentsline}{\par}{}{} % patching contentsline so that it works with partial ToC

\titlecontents{partialsection}[0mm]% partial section style
{\contentsmargin[4mm]{0mm}\fontspec{Fira Sans Book}}%
{{\makebox[3.5mm]{\color{partialTocColor}$\blacktriangleright$\hfill}}\yTocSection}%
{}{}[\vspace*{1mm}]%
\newcommand{\yTocSection}[1]{%
		{#1\hspace*{1mm}(p. \thecontentspage)}%
}

\newcommand{\partialTocTitle}{%
	In Short%
}

\mdfdefinestyle{partialToc}{
	skipabove=\baselineskip,
	skipbelow=\baselineskip,
	innertopmargin=4mm,
	innerbottommargin=\baselineskip,
	innerleftmargin=4mm,
	innerrightmargin=4mm,
	leftmargin=0mm,
	rightmargin=0mm,
	rightline=false,
	topline=false,
	bottomline=false,
	leftline=false,
	linewidth=1mm,
	backgroundcolor=light-gray,
}
\colorlet{partialTocColor}{mainColor}
\DeclareDocumentCommand{\printMarginPartialToc}{O{1} O{\partialTocTitle} }{%
	\marginElement{%
		\begin{mdframed}[style=partialToc]%
			{\fontspec{Fira Sans Book}\Large #2}%\\
			{\color{partialTocColor}\hfill\rule{\linewidth}{.5mm}}\\[2mm]%
			\printcontents{partial}{1}{\setcounter{tocdepth}{#1}\hypersetup{linkcolor=black}}%
		\end{mdframed}%
	}
}```

**Comment by YvesZumbachCMU** [comment]
Hello Wenyin,

I cannot understand your problem fully as I do not see what sectioning commands you issued in your document.

However, from what I see, I think you can solve your problem by adding `\stopcontents` (or maybe `\stopcontents[chapter]`) at the point in your document where you want the mini-toc to stop.

I hope this helps,

Harvey

**Comment by HarveySheppard** [comment]
Hello Wenyin,

I cannot understand your problem fully as I do not see what sectioning commands you issued in your document.

However, from what I see, I think you can solve your problem by adding `\stopcontents` (or maybe `\stopcontents[chapter]`) at the point in your document where you want the mini-toc to stop.

I hope this helps,

Harvey

**Comment by HarveySheppard** [comment]
If you need some more details, you can take a look at the `titlesec` package documentation from which the `\startcontents` and `\stopcontents` commands are coming.

**Comment by WenyinWei** [comment]
Thank you, Harvey!

I have tried adding `\stopcontents` at different places and it didn't work. Even though I change the position where the `\startcontents` is, nothing changes. `\printcontents` seems to be eager to print all contents together. 

A piece of good news is that, after I commented most of the packages one by one, I found some packages are incompatible with `titlesec`, such as `epigraph`, which would cause problems like the aforementioned bug. However, `epigraph` is not the only one which causes the bug, so I totally replace my header file with yours. 

Anyway, thanks for your help and such a dedicated typograph sincerely! It is a fantastic tool to take note. 

Best wishes,
Wenyin

**Comment by HarveySheppard** [comment]
Have you tried printing only the chapter content using `\printcontents[chapter]`?

Anyway, I am glad you like it!
